src/App.jsx

```diff
@@ -14,33 +16,105 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
-
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
-
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
-
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
-
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+const SORT_ALPHABET = 'alphabet';
+const SORT_SIZE = 'size';
+
+export const App = () => {
+  const [goodsOnScreen, setGoodsState] = useState(goodsFromServer);
+  const [sortKind, setSortKind] = useState('');
+  const [isReversed, setReversed] = useState(false);
+
+  function sortAlphabeticaly() {
+    const sortedByLetter = [...goodsOnScreen].sort((item1, item2) =>
+      item1.localeCompare(item2),
+    );
+
+    if (isReversed) {
+      sortedByLetter.reverse();
+    }
+
+    setGoodsState(sortedByLetter);
+    setSortKind(SORT_ALPHABET);
+  }
+
+  function sortByLength() {
+    let sortedByLength;
+
+    if (isReversed) {
+      sortedByLength = [...goodsOnScreen]
+        .reverse()
+        .sort((item1, item2) => item1.length - item2.length)
+        .reverse();
+    } else {
+      sortedByLength = [...goodsOnScreen].sort(
+        (item1, item2) => item1.length - item2.length,
+      );
+    }
+
+    setGoodsState(sortedByLength);
+    setSortKind(SORT_SIZE);
+  }
```

Comment:
It would also be good practice to combine the two sorting functions into one universal one.

Example: 
```
const handleSort = (field) => {}

const handleReverse = () => {}
const handleReset = () => {}

```